src/app/services/map/map.service.ts

```diff
@@ -569,18 +569,119 @@ export class MapService {
+            left: null,
+            right: null,
+            bottom: null
+          },
+          x, y;
+
+        // Iterate over every pixel to find the highest
+        // and where it ends on every axis ()
+        for (i = 0; i < l; i += 4) {
+          if (pixels.data[i + 3] !== 0) {
+            x = (i / 4) % c.width;
+            y = ~~((i / 4) / c.width);
+
+            if (bound.top === null) {
+              bound.top = y;
+            }
+
+            if (bound.left === null) {
+              bound.left = x;
+            } else if (x < bound.left) {
+              bound.left = x;
+            }
+
+            if (bound.right === null) {
+              bound.right = x;
+            } else if (bound.right < x) {
+              bound.right = x;
+            }
+
+            if (bound.bottom === null) {
+              bound.bottom = y;
+            } else if (bound.bottom < y) {
+              bound.bottom = y;
+            }
+          }
+        }
+
+        // Calculate the height and width of the content
+        let trimHeight = bound.bottom - bound.top,
+          trimWidth = bound.right - bound.left,
+          trimmed = ctx.getImageData(bound.left, bound.top, trimWidth, trimHeight);
+
+        copy.canvas.width = trimWidth;
+        copy.canvas.height = trimHeight;
+
+
+        // ctx.globalCompositeOperation = 'multiply';
+        copy.putImageData(trimmed, 0, 0);
+
+        // Return trimmed canvas
+        copy.canvas.toBlob((blob) => {
+          resolve(blob)
+        }, 'image/jpeg')
+      }
+    })
+
+  }
+
   public setSelectedBrowse(url: string, wkt: string) {
     if (!!this.browseImageLayer) {
       this.map.removeLayer(this.browseImageLayer);
     }
-    this.browseImageLayer = this.browseOverlayService.createNormalImageLayer(url, wkt, 'ol-layer', 'current-overlay');
-    this.map.addLayer(this.browseImageLayer);
+    if (!url.endsWith('.tif')) {
+      if(url.includes('OPERA')) {
+        this.trimImage(url).then((imageURL: Blob) => {
+          let url = URL.createObjectURL(imageURL);
+          this.clearBrowseOverlays();
+          this.browseImageLayer = this.browseOverlayService.createImageLayer(url, wkt, 'ol-layer', 'current-overlay');
+          this.map.addLayer(this.browseImageLayer);
+        })
+      } else {
+        this.browseImageLayer = this.browseOverlayService.createNormalImageLayer(url, wkt, 'ol-layer', 'current-overlay');
+        this.map.addLayer(this.browseImageLayer);
+      }
+    } else {
+      fetch(url, {
+        credentials: 'include',
+        redirect: 'follow'
+      }).then(response => response.blob()).then((blob) => {
+        this.browseImageLayer = this.browseOverlayService.createGeotiffLayer(blob, wkt, 'ol-layer', 'current-overlay');
+        let s: any = this.browseImageLayer.getSource().getView()!
+        s.then((thing: any) => {
+          this.map?.getView().fit(thing.extent)
+        })
+        this.map.addLayer(this.browseImageLayer);
+      })
```

Comment:
Is there a way to do this with http module? That's what most/all of are requests are done through elsewhere.